packages/sources/twosigma/src/endpoint/price.ts

```diff
@@ -0,0 +1,182 @@
+    | 'active' // TSS is trading this stock actively
+    | 'inactive' // TSS has not traded this stock recently
+  confidence_interval: number // Confidence interval in quote currency units. e.g. 0.16416514635149188
+  price: number // Price in quote currency units. e.g. 379.64
+}
+
+export type WebSocketEndpointTypes = {
+  Request: {
+    // i.e. { base, quote }
+    // base is the symbol to query, e.g. AAPL
+    // quote is the currency to convert to, e.g. USD
+    Params: PriceEndpointParams
+  }
+  Response: {
+    Data: {
+      result: number
+    }
+    Result: number
+  }
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: WebSocketMessage
+  }
+}
+
+const logger = makeLogger('TwoSigmaPriceWebsocketEndpoint')
+
+export class TwoSigmaWebsocketTransport extends WebSocketTransport<WebSocketEndpointTypes> {
+  // The API works by first sending a message to the server containing the API key
+  // and a list of ticker symbols to subscribe to, after which the server begins
+  // streaming price updates. However, in order to subscribe to a different set of
+  // symbols, the client would need to do so on a new websocket connection as the
+  // server seems to ignore all messages after the first.
+
+  override async streamHandler(
+    context: EndpointContext<WebSocketEndpointTypes>,
+    subscriptions: SubscriptionDeltas<WebSocketEndpointTypes['Request']['Params']>,
+  ): Promise<void> {
+    if (this.wsConnection && (subscriptions.new.length || subscriptions.stale.length)) {
+      logger.debug(
+        `closing WS connection for new subscriptions: ${JSON.stringify(subscriptions.desired)}`,
+      )
+
+      // 'close' is an async method that doesn't return a promise. Rather than
+      // polling and waiting for the connection to close, we can simply set the
+      // variable to null and rely on the parent method to create a new connection.
+      this.wsConnection.close()
+      this.wsConnection = null
+    }
+
+    subscriptions.new = subscriptions.desired
+    subscriptions.stale = []
+
+    super.streamHandler(context, subscriptions)
+  }
+
+  override async sendMessages(
+    _context: EndpointContext<WebSocketEndpointTypes>,
+    subscribes: PriceEndpointParams[],
+    unsubscribes: unknown[],
+  ): Promise<void> {
+    // Send a single message containing the entire set of subscribed symbols rather than
+    // one message per symbol.
+
+    if (unsubscribes.length) {
+      // We explicitly set subscriptions.stale to empty.
+      logger.warn(`unexpected unsubscribes: ${JSON.stringify(unsubscribes)}`)
+    }
+
+    if (!subscribes.length) {
+      // No symbols to subscribe to
+      logger.debug(`nothing to subscribe to, skipping initial message`)
+      return
+    }
+
+    const payload: WebSocketRequest = {
+      api_key: process.env.WS_API_KEY || 'twosigma api key not set',
+      symbols: subscribes.map(buildSymbol).sort(),
+    }
+    this.wsConnection.send(this.serializeMessage(payload))
+  }
+}
```

Comment:
We already set this in the previous method, once the framework version is updated this should not be necessary

Reply:
I'm not sure I follow why this won't be necessary. The overridden `streamHandler` does set subscriptions.new to the entire set of tickers to subscribe to, but the issue is that `sendMessages` on its own will send 1 msg per ticker which won't work. We need to only send 1 msg for all the tickers.

Reply:
Ah, you're right! I was thinking of a previous version (or maybe not even, I've been ooo for a while 😅), this is correct 👍